File: 08/main.pl
use v5.38;
use List::Util qw(sum);

# Get lines from file
open my $fh, "<", "input.txt";
my @lines; while (<$fh>) {chomp; push(@lines, $_)} close $fh;

#// ----- TEST DATA -----
# @lines = (
#     '""',
#     '"abc"',
#     '"aaa\"aaa"',
#     '"\x27"',
# );

my $original_lenghts = sum(map(length, @lines));                # Sumarize len of all lines
my $escaped_lengths = sum(map({length(eval($_))} @lines));      # Sumarize len of all lines after eval-ing them (interpreting escapes)
my $extra_escaped = sum(map({length('"'.($_ =~ s/(\W)/\\$1/gr).'"')} @lines));  # Sumarize len of all lines after applying new escaping

say "Part1: " . ($original_lenghts - $escaped_lengths);
say "Part2: " . ($extra_escaped - $original_lenghts);

File: 08/main.py
import re
with open("input.txt") as file:
    lines = file.read().strip().splitlines()

olen = sum(len(l) for l in lines)
print(f"Original line len: {olen}")

p1 = olen - sum(len(bytes(l[1:-1], 'ascii').decode('unicode_escape')) for l in lines)
p2 = sum(len('"' + re.sub(r'([\"\'\\])', r'\\\1', l) + '"') for l in lines) - olen

print(f"Part1: {p1}")
print(f"Part2: {p2}")
